src/Microsoft.Identity.Web.Certificateless/ManagedIdentityClientAssertion.cs

```diff
@@ -14,12 +15,14 @@ namespace Microsoft.Identity.Web
     public class ManagedIdentityClientAssertion : ClientAssertionProviderBase
     {
         private readonly TokenCredential _credential;
+        private readonly string _tokenExchangeUrl;
 
         /// <summary>
         /// See https://aka.ms/ms-id-web/certificateless.
         /// </summary>
         /// <param name="managedIdentityClientId">Optional ClientId of the Managed Identity or Workload Identity</param>
-        public ManagedIdentityClientAssertion(string? managedIdentityClientId)
+        /// <param name="tokenExchangeUrl">Optional token exchange resource url. Default value is "api://AzureADTokenExchange/.default".</param>
+        public ManagedIdentityClientAssertion(string? managedIdentityClientId, string tokenExchangeUrl = CertificatelessConstants.DefaultTokenExchangeUrl)
```

Comment:
I suggest adding a separate overload with 2 arguments (no optional args). Optional arguments in public APIs can make versioning tricky and result in binary and behavioral breaking changes.

https://haacked.com/archive/2010/08/10/versioning-issues-with-optional-arguments.aspx/
https://haacked.com/archive/2010/08/12/more-optional-versioning-fun.aspx/
https://stackoverflow.com/questions/1456785/a-definitive-guide-to-api-breaking-changes-in-net
https://stackoverflow.com/questions/12670735/is-replacing-an-optional-parameter-with-overloads-a-breaking-change

Reply:
Love this idea - done!